var/spack/repos/builtin/packages/catch/package.py

```diff
@@ -60,9 +60,22 @@ def cmake(self, spec, prefix):
     def build(self, spec, prefix):
         pass
 
-    @when('+single_header')
+    @when('@1.0.0:1.99.99+single_header')
     def install(self, spec, prefix):
         mkdirp(prefix.include)
         install(join_path('single_include', 'catch.hpp'), prefix.include)
         # fakes out spack so it installs a module file
         mkdirp(join_path(prefix, 'bin'))
+
+    @when('@2.0.0:2.4.0+single_header')
+    def install(self, spec, prefix):
+        mkdirp(prefix.include)
```

Comment:
This should be `prefix.include.catch2`. It won't crash as is, but it will rename the `catch.hpp` file to `catch2` instead of placing it in a directory with that name.

Reply:
Wait, the doc string here is not trivial.

As I added above in the comments, it was changed in 2.3.0+:
`releases 2.3.0+ changed to "catch2/catch.hpp" header`
and is as such also documented in the change logs: https://github.com/catchorg/Catch2/releases/tag/v2.3.0

Reply:
I think that diff should be better:
```diff
diff --git a/var/spack/repos/builtin/packages/catch/package.py b/var/spack/repos/builtin/packages/catch/package.py
index 0db27a853..557f135bf 100644
--- a/var/spack/repos/builtin/packages/catch/package.py
+++ b/var/spack/repos/builtin/packages/catch/package.py
@@ -63,6 +63,9 @@ class Catch(CMakePackage):
     @when('+single_header')
     def install(self, spec, prefix):
         mkdirp(prefix.include)
-        install(join_path('single_include', 'catch.hpp'), prefix.include)
+        if spec.satisfies('@2.3.0:'):
+            install_tree('single_include', prefix.include)
+        else:
+            install(join_path('single_include', 'catch.hpp'), prefix.include)
         # fakes out spack so it installs a module file
         mkdirp(join_path(prefix, 'bin'))
```